sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -653,5 +652,81 @@ def Transport(host, connect_timeout=None, ssl=False, **kwargs):
+        transport = WebSocketTransport
     transport = SSLTransport if ssl else TCPTransport
     return transport(host, connect_timeout=connect_timeout, ssl=ssl, **kwargs)
+
+class WebSocketTransport(_AbstractTransport):
+    def __init__(self, host, port=WEBSOCKET_PORT, connect_timeout=None, ssl=None, **kwargs
+        ):
+        self.sslopts = ssl if isinstance(ssl, dict) else {}
+        self._connect_timeout = connect_timeout
+        self._host = host
+        super().__init__(
+            host, port, connect_timeout, **kwargs
+            )
+        self.ws = None
+        self._http_proxy = kwargs.get('http_proxy', None)
+
+    def connect(self):
+        http_proxy_host, http_proxy_port, http_proxy_auth = None, None, None
+        if self._http_proxy:
+            http_proxy_host = self._http_proxy['proxy_hostname']
+            http_proxy_port = self._http_proxy['proxy_hostname']
+            username = self._http_proxy.get('username', None)
+            password = self._http_proxy.get('password', None)
+            if username or password:
+                http_proxy_auth = (username, password)
+        try:
+            from websocket import create_connection
+            self.ws = create_connection(
+                url="wss://{}/$servicebus/websocket/".format(self._host),
+                subprotocols=[AMQP_WS_SUBPROTOCOL],
+                timeout=self._connect_timeout,
+                skip_utf8_validation=True,
+                sslopt=self.sslopts,
+                http_proxy_host=http_proxy_host,
+                http_proxy_port=http_proxy_port,
+                http_proxy_auth=http_proxy_auth
+            )
+        except ImportError:
+            raise ValueError("Please install websocket-client library to use websocket transport.")
+
+    def _read(self, n, buffer=None, **kwargs): # pylint: disable=unused-arguments
+        """Read exactly n bytes from the peer."""
+        
+        length = 0
+        view = buffer or memoryview(bytearray(n))
+        nbytes = self._read_buffer.readinto(view)
+        length += nbytes
+        n -= nbytes
+        while n:
+            data = self.ws.recv()
+
+            if len(data) <= n:
+                view[length: length + len(data)] = data
+                n -= len(data)
+            else:
+                view[length: length + n] = data[0:n]
+                self._read_buffer = BytesIO(data[n:])
+                n = 0
+        return view
+
+    def _shutdown_transport(self):
+        """Do any preliminary work in shutting down the connection."""
+        self.ws.close()
+
+    def _write(self, s):
+        """Completely write a string to the peer."""
+        try:
+            """
+            ABNF, OPCODE_BINARY = 0x2
+            See http://tools.ietf.org/html/rfc5234
+            http://tools.ietf.org/html/rfc6455#section-5.2
+            """
+            from websocket import ABNF
+            self.ws.send(s, opcode=ABNF.OPCODE_BINARY)
```

Comment:
I don't think we want to handle import error in the write method, it would cause slowness, as write will be called millions of times for sending.
we can just but the value 0x2.

Reply:
using the `send_binary` convinience